h2o-py/tests/testdir_algos/stackedensemble/pyunit_stackedensemble_custom_metalearner.py

```diff
@@ -0,0 +1,122 @@
+    stack_gbm = H2OStackedEnsembleEstimator(base_models=[my_gbm, my_rf], metalearner_algorithm="gbm",
+                                            metalearner_params = gbm_params)
+    stack_gbm.train(x=x, y=y, training_frame=train)
+    # Check that metalearner_algorithm is a default GBM
+    assert(stack_gbm.params['metalearner_algorithm']['actual'] == "gbm")
+    # Check that the metalearner is default GBM
+    meta_gbm = h2o.get_model(stack_gbm.metalearner()['name'])
+    assert meta_gbm.algo == "gbm"
+    assert meta_gbm.params['ntrees']['actual'] == 100
+    assert meta_gbm.params['max_depth']['actual'] == 6
+
+
+    # Train a stacked ensemble & metalearner_algorithm "drf"; check that metalearner_algorithm works with CV
+    stack_drf = H2OStackedEnsembleEstimator(base_models=[my_gbm, my_rf], metalearner_algorithm="drf", metalearner_nfolds=3,
+                                            metalearner_params = drf_params)
+    stack_drf.train(x=x, y=y, training_frame=train)
+    # Check that metalearner_algorithm is a default RF
+    assert(stack_drf.params['metalearner_algorithm']['actual'] == "drf")
+    # Check that CV was performed
+    assert(stack_drf.params['metalearner_nfolds']['actual'] == 3)
+    meta_drf = h2o.get_model(stack_drf.metalearner()['name'])
+    assert meta_drf.algo == "drf"
+    assert meta_drf.params['nfolds']['actual'] == 3
+    assert meta_drf.params['ntrees']['actual'] == 100
+    assert meta_drf.params['max_depth']['actual'] == 6
+
+
+    # Train a stacked ensemble & check that metalearner_algorithm "glm" works
+    stack_glm = H2OStackedEnsembleEstimator(base_models=[my_gbm, my_rf], metalearner_algorithm="glm",
+                                            metalearner_params = glm_params)
+    stack_glm.train(x=x, y=y, training_frame=train)
+    # Check that metalearner_algorithm is a default GLM
+    assert(stack_glm.params['metalearner_algorithm']['actual'] == "glm")
+    # Check that the metalearner is default GLM
+    meta_glm = h2o.get_model(stack_glm.metalearner()['name'])
+    assert meta_glm.algo == "glm"
+    assert meta_glm.params['alpha']['actual'] == [0.0]
+    assert meta_glm.params['lambda']['actual'] == [0.0]
+
+
+    # Train a stacked ensemble & check that metalearner_algorithm "deeplearning" works
+    stack_dl = H2OStackedEnsembleEstimator(base_models=[my_gbm, my_rf], metalearner_algorithm="deeplearning",
+                                           metalearner_params = dl_params)
+    stack_dl.train(x=x, y=y, training_frame=train)
+    # Check that metalearner_algorithm is a default DNN
+    assert(stack_dl.params['metalearner_algorithm']['actual'] == "deeplearning")
+    # Check that the metalearner is default DNN
+    meta_dl = h2o.get_model(stack_dl.metalearner()['name'])
+    assert(meta_dl.algo == "deeplearning")
+    assert meta_dl.params['hidden']['actual'] == [32, 32, 32]
+    assert meta_dl.params['epochs']['actual'] == 20.0
+
+
```

Comment:
I wonder if we handle the following cases correctly (maybe requires a test?):

- pass in incorrect param names or values
- pass in `metalearner_params` when `metaleraner_algorithm = "AUTO"`.

Reply:
I can test this scenario

Reply:
Basically it will be ignored if set to `AUTO` and you will hit an illegal argument error if you pass in incorrect params.

Reply:
Sounds good.  No need to add additional tests at this time.